qsiprep/cli/parser.py

```diff
@@ -99,6 +99,23 @@ def _min_one(value, parser):
             raise parser.error("Argument can't be less than one.")
         return value
 
+    def _int_or_auto(value, parser):
+        """Ensure an argument is an integer or 'auto'."""
+        if value == 'auto':
+            return value
+        try:
+            value = int(value)
+        except ValueError as exc:
+            raise parser.error('Argument must be an integer or "auto".') from exc
+
+        if value < 1:
+            raise parser.error('Argument must be greater than zero.')
+
+        if value % 2 == 0:
+            raise parser.error('Argument must be an odd integer.')
```

Comment:
We don't use the value for patch2self at all though, so the only time `--dwi-denoise-window` has an effect is if you use dwidenoise.

Reply:
ah, so this doesn't even matter here

Reply:
Yeah, the parser is currently going to raise an error if a user provides a non-auto value and _isn't_ using dwidenoise, since it won't have an effect.